### PR TITLE
rpc: Clarify decodescript RPCResult doc

### PR DESCRIPTION
**Pull request by maflcko**
* Remove `"hex"` from the decodescript RPCResult doc
* Add `"segwit`" to the doc

Follow up to a6099ef319a73e2255dca77065600abb22c4f5f8 and 4f933b3d23010d3b03998460290faed97cd6f236

### CODE REVIEW

File: src/rpc/rawtransaction.cpp [hunk 1]
```diff
@@ -610,31 +610,42 @@ static UniValue decoderawtransaction(const JSONRPCRequest& request)
...
                 RPCResult{
             "{\n"
-            "  \"asm\":\"asm\",   (string) Script public key\n"
-            "  \"hex\":\"hex\",   (string) hex-encoded public key\n"
-            "  \"type\":\"type\", (string) The output type\n"
-            "  \"reqSigs\": n,    (numeric) The required signatures\n"
-            "  \"addresses\": [   (json array of string)\n"
-            "     \"address\"     (string) bitcoin address\n"
+            "  \"asm\":\"asm\",          (string) Script public key\n"
+            "  \"type\":\"type\",        (string) The output type\n"
```
Comment (ryanofsky) [inline src/rpc/rawtransaction.cpp]: In commit "rpc: Clarify decodescript RPCResult doc" (faad33ff15398159115afd4799e3c97ac3df2a3a)

Might be nice to list possible type strings here.

**Review by ryanofsky — APPROVED** [review]
utACK fa3caa1666a9baa856e92398a890aaddbdd98f7c. Looks like segwit field was added in #12321 and never documented.

File: src/rpc/rawtransaction.cpp (hunk 1 above)
Comment (MarcoFalke) [reply src/rpc/rawtransaction.cpp]: Done

**Review by ryanofsky — APPROVED** [review]
utACK fa926ec24fb3d07de32bd8f67a297f9e4f0822a6. Only change since last review is listing possible output types in the help string using a new `GetAllOutputTypes` function

**Pull request merged**